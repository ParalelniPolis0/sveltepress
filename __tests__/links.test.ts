import { describe, expect, it } from 'vitest'
import { compile } from 'mdsvex'
import links from '../src/markdown/links'

const md = `
## Foo

[foo](https://www.google.com/)
[bar](/foo/bar)

`

describe('links', () => {
  it('simple', async () => {
    const r = await compile(md, {
      remarkPlugins: [links, () => (tree) => {
        expect(JSON.stringify(tree, null, 2)).toMatchSnapshot()
      }],
    })
    expect(r).toMatchInlineSnapshot(`
      {
        "code": "
      <h2>Foo</h2>
      <p><Link to=\\"https://www.google.com/\\" label=\\"foo\\" />
      <Link to=\\"/foo/bar\\" label=\\"bar\\" /></p>
      ",
        "data": {
          "anchors": [],
        },
        "map": "",
      }
    `)
  })
})
