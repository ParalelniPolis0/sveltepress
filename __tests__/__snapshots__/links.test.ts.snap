// Vitest Snapshot v1

exports[`links > simple 1`] = `
"{
  \\"type\\": \\"root\\",
  \\"children\\": [
    {
      \\"type\\": \\"heading\\",
      \\"depth\\": 2,
      \\"children\\": [
        {
          \\"type\\": \\"text\\",
          \\"value\\": \\"Foo\\",
          \\"position\\": {
            \\"start\\": {
              \\"line\\": 2,
              \\"column\\": 4,
              \\"offset\\": 4
            },
            \\"end\\": {
              \\"line\\": 2,
              \\"column\\": 7,
              \\"offset\\": 7
            },
            \\"indent\\": []
          }
        }
      ],
      \\"position\\": {
        \\"start\\": {
          \\"line\\": 2,
          \\"column\\": 1,
          \\"offset\\": 1
        },
        \\"end\\": {
          \\"line\\": 2,
          \\"column\\": 7,
          \\"offset\\": 7
        },
        \\"indent\\": []
      }
    },
    {
      \\"type\\": \\"paragraph\\",
      \\"children\\": [
        {
          \\"type\\": \\"html\\",
          \\"value\\": \\"<Link to=\\\\\\"https://www.google.com/\\\\\\" label=\\\\\\"foo\\\\\\" />\\"
        }
      ],
      \\"position\\": {
        \\"start\\": {
          \\"line\\": 4,
          \\"column\\": 1,
          \\"offset\\": 9
        },
        \\"end\\": {
          \\"line\\": 4,
          \\"column\\": 31,
          \\"offset\\": 39
        },
        \\"indent\\": []
      }
    }
  ],
  \\"position\\": {
    \\"start\\": {
      \\"line\\": 1,
      \\"column\\": 1,
      \\"offset\\": 0
    },
    \\"end\\": {
      \\"line\\": 6,
      \\"column\\": 1,
      \\"offset\\": 41
    }
  }
}"
`;
